### PR TITLE
Add backend for exact match search

### PR DESCRIPTION
**Pull request by anthonytate**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+/** Servlet to test set-up */
+@WebServlet("/skillsearch")
+public class SearchServlet extends HttpServlet {
+  private ArrayList<Skill> list;
+  private String query;
+
+  private void initializeDatabase() {
+    // To set the default application credentials you must run the following command:
+    // GOOGLE_APPLICATION_CREDENTIALS="path/to/firebase-key.json"
+    // and add that file to the .gitignore if it's location is different than the one that's
```
Comment (hack-heart) [inline src/main/java/com/google/sps/servlets/SearchServlet.java:50]: nit: its location

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    } catch (InterruptedException e) {
+      System.err.println(e);
+    }
+
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(list));
+  }
+
+  class Worker implements Runnable {
```
Comment (hack-heart) [inline src/main/java/com/google/sps/servlets/SearchServlet.java:89]: Should this class be in a different file?

File: src/main/java/com/google/sps/data/Skill.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package com.google.sps.data;
+
+import java.util.ArrayList;
+
+public class Skill {
+  private String category;
+  private String desc;
+  private ArrayList<String> users;
+
+  public Skill() {}
```
Comment (hack-heart) [inline src/main/java/com/google/sps/data/Skill.java:10]: No need for the empty constructor.

Comment (anthonytate) [reply src/main/java/com/google/sps/data/Skill.java:10]: It's required by the API that the class that gets passed in for getValue() (in SearchServlet.java) has a default constructor with no arguments. It's how they did it in the example here: https://firebase.google.com/docs/reference/admin/java/reference/com/google/firebase/database/DataSnapshot#public-t-getvalue-classt-valuetype

I'm not sure if Java creates default constructors automatically though. If that's the case I can take it out but I just followed the example they used

Comment (hack-heart) [reply src/main/java/com/google/sps/data/Skill.java:10]: Sweet. Disregard the comment

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+public class SearchServlet extends HttpServlet {
+  private ArrayList<Skill> searchResults;
+  private String query;
+
+  private void initializeDatabase() {
+    // To set the default application credentials you must run the following command:
+    // GOOGLE_APPLICATION_CREDENTIALS="path/to/firebase-key.json"
+    // and add that file to the .gitignore if its location is different than the one that's
+    // currently there
+    // You must do this for every session because the value will be deleted on session end
```
Comment (Ali09) [inline src/main/java/com/google/sps/servlets/SearchServlet.java:52]: Small nit: add periods.

**Pull request merged**